### PR TITLE
[#465] Handled multiline dependencies

### LINKED ISSUE #465: requirements-txt-fixer does not handle multiline dependencies
Given:
```
distlib==0.3.0 \
    --hash=sha256:2e166e231a26b36d6dfe35a48c4464346620f8645ed0ace01ee31822b288de21
appdirs==1.4.3 \
    --hash=sha256:9e5896d1372858f8dd3344faf4e5014d21849c756c8d5701f78f8a103b372d92 \
    --hash=sha256:d8b24664561d0d34ddfaec54636d502d7cea6e29c3eaf68f3df6180863e2166e
filelock==3.0.12 \
    --hash=sha256:18d82244ee114f543149c66a6e0c14e9c4f8a1044b5cdaadd0f82159d6a6ff59 \
    --hash=sha256:929b7d63ec5b7d6b71b0fa5ac14e030b3f70b75747cef1b10da9b879fef15836
```

requirements-txt-fixer will move all `--hash` lines on top of the file.

### PR DESCRIPTION
**Pull request by aniketbhatnagar**
Resolves #465

### CODE REVIEW

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 1]
```diff
@@ -34,6 +34,15 @@ def __lt__(self, requirement: 'Requirement') -> int:
         else:
             return self.name < requirement.name
 
+    def is_complete(self) -> bool:
+        return self.value is not None and not self.value.strip().endswith(b'\\')
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py:38]: this should probably a little more specific, I would do `.rstrip(b'\r\n')` -- that way something that has trailing whitespace but a backslash at the end isn't considered a continuation

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 2]
```diff
@@ -34,6 +34,15 @@ def __lt__(self, requirement: 'Requirement') -> int:
...
+
+    def append_value(self, value: bytes):
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py:40]: the linting is saying you need `-> None` on this function

File: tests/requirements_txt_fixer_test.py [hunk 3]
```diff
@@ -50,6 +50,24 @@
             FAIL,
             b'Django\nijk\ngit+ssh://git_url@tag#egg=ocflib\n',
         ),
+        (
+            b'b==1.0.0\n'
+            b'c=2.0.0 \\ \n'
```
Comment (asottile) [inline tests/requirements_txt_fixer_test.py:55]: whitespace after a backslash I believe is not valid for a contiuation? 🤔

File: pre_commit_hooks/requirements_txt_fixer.py (hunk 1 above)
Comment (aniketbhatnagar) [reply pre_commit_hooks/requirements_txt_fixer.py]: Right, I missed that. Fixed

File: pre_commit_hooks/requirements_txt_fixer.py (hunk 2 above)
Comment (aniketbhatnagar) [reply pre_commit_hooks/requirements_txt_fixer.py]: fixed

File: tests/requirements_txt_fixer_test.py (hunk 3 above)
Comment (aniketbhatnagar) [reply tests/requirements_txt_fixer_test.py]: Ya, fixed test case

**Comment by asottile** [comment]
just a small tip for next time, if you put "Resolves #XXX" in your PR description it will auto-close the issue upon merge

**Pull request merged**